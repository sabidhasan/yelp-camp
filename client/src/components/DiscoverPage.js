import React from 'react'
import { Switch, Route } from 'react-router-dom'
import { provinces } from '../helpers/helpers'
import DiscoverGoogleMap from './DiscoverGoogleMap'

class Discover extends React.Component {
 constructor(props) {
   super(props)
   if (!provinces[this.props.match.params.province]) window.location = '/';
   const province = {
     fullName: provinces[this.props.match.params.province],
     shortName: this.props.match.params.province
   }
   this.state = {
     province: province,
     campgrounds: null
   }
 }

 componentDidMount() {
   fetch(`/campground?province=${this.state.province.shortName}`)
    .then(res => res.json())
    .then(provCG => {
      this.setState({campgrounds: provCG})
    });
 }

 render() {
   return (
   <div className='discover'>
     <h1>This is the single page for {this.state.province.fullName}</h1>

     <div>
       <DiscoverGoogleMap
         googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyC4R6AN7SmujjPUIGKdyao2Kqitzr1kiRg&v=3.exp&libraries=geometry,drawing,places"
         loadingElement={<div className='map-container' />}
         containerElement={<div className='map-container' />}
         mapElement={<div className='map-container' />}
         lat={this.state.lat}
         lon={this.state.lon}
       />
     </div>
   </div>
)
   // return
 }
}

export default Discover
