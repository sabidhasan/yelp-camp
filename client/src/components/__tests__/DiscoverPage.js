import React from 'react'
import toJson from 'enzyme-to-json'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import DiscoverPage from '../DiscoverPage'

configure({ adapter: new Adapter() });

describe('<DiscoverPage />', () => {
  let component;
  let props;
  let context;

  beforeEach(() => {
    props = {
      match: {params: {province: 'ab'}},
      history: [],
    }
    component = shallow(<DiscoverPage { ...props } />, {context: {userLocation: {}}})
    // Fake the AJAX call and set component state directly
    component.setState({
      campgrounds: [
        {id: 1}, {id: 2}, {id: 3}
      ]
    })
  })

  it('Component should be defined', () => {
    expect(component).toBeDefined();
  });

  it('Null is returned when no AJAX call is made', () => {
    component.setState({ campgrounds: null })
    expect(component.html()).toBe(null);
  })

  it('If an invalid province is given, redirect to home page', () => {
    // Make an invalid province
    props.match.params.province = 'FF'
    component = shallow(<DiscoverPage { ...props } />)
    expect(props.history[0]).toBe('/');
  })

  it('Header element displays province name', () => {
    expect(component.find('.Discover__province').text()).toBe('alberta')
  })

  it('Discover length has correct length', () => {
    expect(component.find('.Discover__length').text()).toBe('3')
  })

  it('No DiscoverCampgroundTile exists when component loads (no campground selected)', () => {
    expect(component.find('DiscoverCampgroundTile')).toHaveLength(0)
  })

  it('When setSelected is called with valid CG id, DiscoverCampgroundTile is shown ', () => {
    component.instance().setSelected(2);
    expect(component.find('DiscoverCampgroundTile')).toHaveLength(1)
  })

  it('Calling setSelected with invalid CG id, hides DiscoverCampgroundTile', () => {
    component.instance().setSelected(5);
    expect(component.find('DiscoverCampgroundTile')).toHaveLength(0)
  })

  it('DiscoverPage component snapshot matches what is expected', () => {
    expect(toJson(component)).toMatchSnapshot();
  })
});
