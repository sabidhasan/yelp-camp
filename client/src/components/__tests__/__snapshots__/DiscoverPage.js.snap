// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DiscoverPage /> DiscoverPage component snapshot matches what is expected 1`] = `
<div
  className="Discover"
>
  <div
    className="Discover__title"
  >
    <h1
      className="Discover__header"
    >
      Discover campgrounds in 
      <span
        className="Discover__province"
      >
        alberta
      </span>
    </h1>
    <p>
      We have 
      <span
        className="Discover__length"
      >
        3
      </span>
       campgrounds in 
      alberta
      , of which
       3
       are
       
       shown on the map
    </p>
  </div>
  <div
    className="Discover__content"
  >
    <div
      className="Discover__map google-map"
    >
      <Wrapper
        campgrounds={
          Array [
            Object {
              "id": 1,
            },
            Object {
              "id": 2,
            },
            Object {
              "id": 3,
            },
          ]
        }
        setSelected={[Function]}
      />
    </div>
  </div>
</div>
`;
